### PR TITLE
Add posthog analytics

### PR DESCRIPTION
**Pull request by oredavids**
Adds Posthog to [try out their product analytics.](https://gruntwork-io.slack.com/archives/C76UJFUEL/p1641926893098400)

### CODE REVIEW

**Comment by oredavids** [comment]
@ebeneliason @eak12913 I need a review & approval so we can get this merged.

File: scripts/push-to-s3-prod.sh [hunk 1]
```diff
@@ -6,6 +6,9 @@ export CONFIG_SITE_URL="https://docs.gruntwork.io"
 export ALGOLIA_INDEX_NAME="docs_site_prod"
 export ALGOLIA_API_KEY="49a495ba4c210780a28feed306d05522" # This is a search only key. It is safe to check in.
 
+export CONFIG_POSTHOG_API_KEY="phc_zJYg43ctbPl1BzsnODjop1op9KSsctiGm6DNNt3zHZT"
+export CONFIG_POSTHOG_APP_URL="https://docs.gruntwork.io"
```
Comment (eak12913) [inline scripts/push-to-s3-prod.sh:10]:
```suggestion
export CONFIG_POSTHOG_APP_URL="$CONFIG_SITE_URL"
```

**Review by ebeneliason — APPROVED** [review]
Reviewed with @eak12913.

File: scripts/push-to-s3-prod.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ export ALGOLIA_INDEX_NAME="docs_site_prod"
 export ALGOLIA_API_KEY="49a495ba4c210780a28feed306d05522" # This is a search only key. It is safe to check in.
 
 export CONFIG_POSTHOG_API_KEY="phc_zJYg43ctbPl1BzsnODjop1op9KSsctiGm6DNNt3zHZT"
-export CONFIG_POSTHOG_APP_URL="https://docs.gruntwork.io"
```
Comment (oredavids) [inline scripts/push-to-s3-prod.sh]: I considered this but worried it would break stage because the posthogConfig would exist but there will no apiKey.

**Pull request merged**